libs/frontend/feature-build/src/components/build-render/ProseMirrorEditor.vue

```diff
@@ -104,13 +104,8 @@ watch(gradientColorStyle, () => {
 })
 
 onBeforeUnmount(() => {
-  if (editor.value?.editView) {
-    if (editor.value.editView.state.doc.content.size <= 2) {
-      editComponent(component.value, { content: '' })
-    }
-    editor.value.editView?.destroy()
-    editor.value.editView = undefined
-  }
+  editor.value.editView?.destroy()
+  editor.value.editView = undefined
```

Comment:
I noticed that the [fix](https://github.com/samatechtw/pubstudio-builder/pull/84#discussion_r1400702449) for the prosemirror content cleared issue would make the component keep getting focused on unmount due to the "auto focus on affected component" feature in all commands. We're unable to select another component when we clear out the text in a prosemirror editor.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/a455e5bc-bc54-4ae1-9235-9d97e6510f8b

After trying out for a while I think it's easier to just use update the content of the target component to `undefined` in the transaction of a prosemirror editor. I adjusted `editorStateToHtml` to make it work.

Reply:
Makes sense, I probably should have thought of that in the first place!